### PR TITLE
IE-345/organization-url-check + add url parser

### PR DESCRIPTION
**Pull request by vuthikxkol**
Add url check for ping operation and unit tests.

### CODE REVIEW

**Comment by vuthikxkol** [comment]
@dspasojevic @dbalmain Ready for review

File: src/scripts/services/okta-service.ts [hunk 1]
```diff
@@ -66,24 +81,22 @@ export const validateCredentials = (
...
-      (error: unknown) => {
-        // eslint-disable-next-line functional/no-conditional-statement
-        if (error instanceof Error) {
-          return error;
-        }
-        return new Error('Failed to ping okta server.', {
-          cause: error,
-        });
-      }
+    parseUrl(url),
```
Comment (dspasojevic) [inline src/scripts/services/okta-service.ts:87]: **Issue:** this should happen while processing the command line arguments (and happen consistently for all commands that accept an org URL).

File: src/schema.ts [hunk 2]
```diff
@@ -1,7 +1,21 @@
...
+  .startsWith('https://', `URL must start with [${urlStart}].`)
+  //.co will cause fetch to time out
+  .endsWith('.okta.com', `URL must end with [${urlEnd}].`)
+  .min(
+    urlStart.length + urlEnd.length + 1,
+    'Domain name must be at least 1 character long.'
+  )
+  .refine(
+    (url) => !url.endsWith('-admin.okta.com'),
+    'Organisation URL should not be the admin URL. Please remove "-admin" and try again.'
```
Comment (dspasojevic) [inline src/schema.ts:20]: **Praise:** these are great messages.

File: src/schema.ts [hunk 3]
```diff
@@ -1,7 +1,21 @@
...
   status: z.number(),
 });
-
-export type OktaAPIError = z.infer<typeof oktaAPIError>;
+const urlStart = 'https://';
+const urlEnd = '.okta.com';
+export const urlSchema = z
+  .string()
+  .url()
+  .startsWith('https://', `URL must start with [${urlStart}].`)
```
Comment (dspasojevic) [inline src/schema.ts:11]:
```suggestion
  .startsWith(urlStart, `URL must start with [${urlStart}].`)
```

File: src/scripts/services/okta-service.ts (hunk 1 above)
Comment (vuthikxkol) [reply src/scripts/services/okta-service.ts]: added in commit: 221668d5d0b826386b4476d753022859713b8b0c

File: src/schema.ts (hunk 2 above)
Comment (vuthikxkol) [reply src/schema.ts:20]: Thank you

File: src/schema.ts (hunk 3 above)
Comment (vuthikxkol) [reply src/schema.ts]: modified in commit: 221668d5d0b826386b4476d753022859713b8b0c

File: src/scripts/ping.ts [hunk 4]
```diff
@@ -67,22 +71,28 @@ export default (
...
     }) => {
-      const client = oktaReadOnlyClient({ ...args });
-
-      const { clientId, organisationUrl } = args;
-      const result = await validateOktaServerAndCredentials(
-        client,
-        clientId,
-        organisationUrl
+      const { organisationUrl, clientId } = args;
+      const result = await parseUrlWrapper(organisationUrl, (url: string) =>
```
Comment (dspasojevic) [inline src/scripts/ping.ts:83]: **Issue:** `organisationUrl` is defined in one place (`src/index.ts`). It is a bit of a code smell to have it defined in one place but then parsed in all the commands.

Is there a way for you to parse the `organisationUrl` in one place?

Comment (vuthikxkol) [reply src/scripts/ping.ts]: modified in commit: fef29bd8f6e2900782da9d97661727803150f786
This is a much better way to do things

**Comment by vuthikxkol** [comment]
@dspasojevic @dbalmain Ready for re-review

**Review by dspasojevic — APPROVED** [review]

**Pull request merged**